### PR TITLE
Conan!

### LINKED ISSUE #230: Investigate the use of Conan instead of submodules for third-party dependencies
https://conan.io/

It's already being used for some Cesium-internal C++ projects.

The main goal being to reduce cesium-native git clone and compilation time.

### LINKED ISSUE #248: Publish cesium-native to conan and vcpkg
Once cesium-native reaches some level of stability we should consider publishing it to conan and vcpkg like how we publish CesiumJS to npm.

### PR DESCRIPTION
**Pull request by kring**
This PR does a bunch of things:

* Drastically simplifies the CMake setup:
  * Extracted all the utility/helper CMake code to `tools/cesium.cmake`.
  * Each library's CMakeLists.txt is now a proper CMake project that can be used standalone.
  * The CMake code in each configuration has been drastically simplified. It mostly just calls `cesium_library` with its name and dependencies. That function takes care of everything else.
  * Our CMake still does not depend on Conan at all. Dependencies are found by using `find_package`.
* Eliminates all the git submodules.
* Adds a `cesium-native.yml` file in the root, and a `library.yml` in each library directory. These files capture information about cesium-native and the libraries that it contains, including third-party dependencies and their versions.
* Added `tools/automate.py` to do various things, like generate Conan recipes for the libraries from the descriptors in `library.yml` and `cesium-native.yml`. Strangely enough this is the first Python program I've ever written, so hopefully it's not too terrible. I used Python because Conan is pretty all-in on it, and it was pretty easy to do. See README.md.
* Switched from modp_b64 (copied out of Chromium) to base64 (from Conan Center).
* Added recipes for the three remaining third-party libraries that are not yet on Conan Center: csprng, picosha2, and s2geometry.
* Added VSCode CMake Kits to use toolchain files generated by Conan.
* Added instructions for installing dependencies with Conan and building Conan packages.

Fixes #230 
Fixes #248

This PR is a draft until there's a rock-solid cesium-unreal PR to go with it.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -21,29 +23,54 @@ jobs:
...
+        - pip install conan
       before_script:
-        - mkdir -p build
-        - cd build
-        - cmake .. -A x64
+        - py -3 ./tools/automate.py conan-export-recipes
+        - py -3 ./tools/automate.py generate-library-recipes --editable
+        - py -3 ./tools/automate.py editable on
+        - conan install . --build=outdated --build=cascade -pr:h=tools/conan-profiles/$CONAN_PROFILE -pr:b=tools/conan-profiles/$CONAN_PROFILE -s build_type=Debug
+      after_script:
```
Comment (baothientran) [inline .travis.yml:39]: It would be nice if we can provide a command to automate all those above steps

**Comment by baothientran** [comment]
looks good to me! Nothing comes out as odd

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @kring! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @kring! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @kring! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @kring! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @kring! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @kring! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @kring! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @kring! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @kring! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @kring! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @kring! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @kring! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @kring! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @kring! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @kring! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @kring! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @kring! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @kring! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by kring** [comment]
Closing this old PR as we're heading toward vcpkg at this point.

**Pull request closed without merging**